.devcontainer/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM mcr.microsoft.com/devcontainers/universal:2
+
+ENV GENERATE_SOURCEMAP=false
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install -y \
+    xvfb \
+    xsel \
+    libgtk2.0-0 \
+    libgtk-3-0 \
+    libgbm-dev \
+    libnotify-dev \
+    libnss3 \
+    libxss1 \
+    libasound2 \
+    libxtst6 \
+    xauth
```

Comment:
I'm not sure that this reasoning is correct (or maybe I'm misunderstanding):

> This Dockerfile also uses the [Advanced Package Tool](https://en.wikipedia.org/wiki/APT_(software)) program apt-get to add the prerequisites Cypress requires to run in a Linux Eviorment, [which are outlined here](https://docs.cypress.io/guides/getting-started/installing-cypress#Linux-Prerequisites). We do this at this level as several dependencies/prerequisites, including xsel and xvfb, can't be handled by yarn.

Cypress already uses its own docker image, so it shouldn't be necessary here.

Do we even need a custom Dockerfile? It makes the use of Codespaces very impractical because of how long it takes to build. For comparison, if you open a Codespace in the `main` branch, it only takes a few seconds.

I think you might be able to replace the Dockerfile with a [predefined template](https://docs.github.com/en/codespaces/developing-in-a-codespace/creating-a-codespace-from-a-template), which starts up much faster.

Reply:
That's a good point! I'll look into that and see if that's possible

Reply:
1. Creating a template from an existing repository [requires access to the settings](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository), which as a non-admin of iTwinUI I don't have access to. Similarly I looked into [prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds) also for time reasons and ran into the same issue.

Reply:
I meant using an existing template, rather than creating a new one. Before that though, can you try simply using the linux universal image as-is (instead of custom dockerfile)?

```diff
- "build": {
-   "dockerfile": "Dockerfile"
- },
+ "image": "mcr.microsoft.com/devcontainers/universal:2",
```

Reply:
When using the universal image, the dependency issues that lead me to use a Dockerfile come up: 
![image](https://github.com/iTwin/iTwinUI/assets/148785417/1d365405-1f6c-4e62-bbfb-0336940a5cdf)

I can look into existing templates

Reply:
That error is coming from `a11y` tests, where we do not use a docker container. If you try running the react visual tests for example, it should work fine.

To fix this issue, you can change the `a11y` test configuration to also run in the `cypress:included` container (same as `react-workshop` test configuration).

Reply:
looking at the `cypress.config` for both the `a11y` tests and the `react-workshop` tests, I don't recognize anything relating to containers. Would these configurations be found somehere else?

Reply:
See here: https://github.com/iTwin/iTwinUI/blob/main/apps/react-workshop/scripts/run-tests.js#L8

Reply:
sweet thank you! I had found `run-tests.js` and have been working through understanding the customizations actually. I appreciate the tips

Reply:
I've come to this possible (but not working solution) and I was hoping to get some feedback on whether or not I'm on the right path or not. 

So in the `package.json` of `testing/a11y` I've changed the `test` property to be: `"docker run --rm --entrypoint /bin/bash -v %cd%:/testing/a11y -w /e2e 'cypress/included:13.2.0' run --component "`

This is based on several docs I've read, and the script being run in `run-tests.js`. While there's still some errors I'm working through I just wanted to see if this approach in generally the on the righ track

Reply:
> So in the `package.json` of `testing/a11y` I've changed the `test` property to be: `"docker run --rm --entrypoint /bin/bash -v %cd%:/testing/a11y -w /e2e 'cypress/included:13.2.0' run --component "`

Yeah, that's generally the right approach

Reply:
Resolving this thread based on internal discussion.

> I found some success using this command:
>   ```
>     "test": "docker run --rm --entrypoint /bin/bash -v $PWD:/iui -w /iui  cypress/included:13.2.0 -c \"yarn workspace a11y test:run\"",
>     "test:run": "cypress run --component",
>   ```
> 
> However, there is another issue caused by some optional platform-specific dependencies missing when the host system (responsible for installing deps) is on a different platform. This is a [bug in npm](https://redirect.github.com/npm/cli/issues/4828) itself.
> 
> For the purpose of Codespaces though, we can proceed without a11y testing working. It can be fixed in a future PR.